graphql_compiler/compiler/emit_sql.py

```diff
@@ -348,10 +509,28 @@ def _get_fold_outputs(self, fold_scope_location, all_folded_outputs):
                         intermediate_fold_output_name = FOLD_OUTPUT_FORMAT_STRING.format(
                             fold_output.field)
                         # add array aggregated output column to self._outputs
-                        self._outputs.append(
-                            self._get_array_agg_column(intermediate_fold_output_name,
-                                                       fold_output.field)
-                        )
+                        # add aggregated output column to self._outputs
+                        if isinstance(self._sqlalchemy_compiler, MSDialect):
+                            # MSSQL uses XML PATH aggregation
+                            self._outputs.append(
+                                self._get_mssql_xml_path_column(
```

Comment:
It seems that all non-error paths here end up calling `append()`, and the question is what to append to `self._outputs`. Consider pulling the `self._outputs.append()` call outside of the `if-elif` and instead use the `if-elif` to just construct the expression that gets appended.